apps/tup-ui/src/components/auth/LoginComponent/LoginComponent.tsx

```diff
@@ -59,9 +59,17 @@ const LoginError: React.FC<{ status?: number }> = ({ status }) => {
 
 const LoginComponent: React.FC<LoginProps> = ({ className }) => {
   const location = useLocation();
-  const from = (location.state as { from?: Location })?.from?.pathname || '/';
+  const [searchParams] = useSearchParams();
+
+  // Redirect precedence is 1) last router history location, 2) query string 'from'
+  // parameter, 3) dashboard base route
+  let from = (location.state as { from?: Location })?.from?.pathname ?? '';
+  if (!from) {
+    from = searchParams.get('from') ?? '/dashboard';
```

Comment:
The default `django.contrib.auth.REDIRECT_FIELD_NAME` is `next`. Can you test an interaction with the default `@login_required` decorator?